ipatests/test_webui/test_user.py

```diff
@@ -376,6 +409,249 @@ def fill_password_dialog(self, password, current=None):
         self.wait_for_request(n=3)
         self.assert_no_error_dialog()
 
+    @screenshot
+    def test_login_without_username(self):
+
+        self.init_app(login='', password='xxx123', skip_username=True)
+
+        alert_e = self.find('.alert[data-name="username"]',
+                            By.CSS_SELECTOR)
+        assert 'Username: Required field' in alert_e.text, 'Alert expected'
+        assert self.login_screen_visible()
+
+    @screenshot
+    def test_disable_delete_admin(self):
+        """
+        Test disabling/deleting admin is not allowed
+        """
+        self.init_app()
+        self.navigate_to_entity('user')
+
+        # try to disable admin user
+        self.select_record('admin')
+        self.facet_button_click('disable')
+        self.dialog_button_click('ok')
+        self.assert_last_error_dialog(ERR_ADMIN_DEL, details=True)
+        self.dialog_button_click('ok')
+        self.assert_record('admin')
+
+        # try to delete admin user. Later we are
+        # confirming by keyboard to test also upstream ticket 4097
+        self.select_record('admin')
+        self.facet_button_click('remove')
+        self.dialog_button_click('ok')
+        self.assert_last_error_dialog(ERR_ADMIN_DEL, details=True)
+        actions = ActionChains(self.driver)
+        actions.send_keys(Keys.TAB)
+        actions.send_keys(Keys.ENTER).perform()
+        # self.dialog_button_click('ok')
+        self.wait(0.5)
+        self.assert_record('admin')
+
+    @screenshot
+    def test_add_user_special(self):
+        """
+        Test various add user special cases
+        """
+
+        self.init_app()
+
+        # Test invalid characters (#@*?) in login
+        self.navigate_to_entity('user')
+        self.facet_button_click('add')
+        self.fill_textbox('uid', 'itest-user#')
+        self.assert_field_validation(ERR_INCLUDE)
+        self.fill_textbox('uid', 'itest-user@')
+        self.assert_field_validation(ERR_INCLUDE)
+        self.fill_textbox('uid', 'itest-user*')
+        self.assert_field_validation(ERR_INCLUDE)
+        self.fill_textbox('uid', 'itest-user?')
+        self.assert_field_validation(ERR_INCLUDE)
+        self.dialog_button_click('cancel')
+
+        # Add an user with special chars
+        self.init_app()
```

Comment:
Why `init_app` here? And below?

Reply:
Notes before fix: Those are leftovers from squashing the particular test cases in one test_method.